ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java

```diff
@@ -40,48 +54,87 @@
 import java.util.StringTokenizer;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
-import org.apache.commons.lang3.StringUtils;
-import org.jline.reader.EndOfFileException;
-import org.jline.terminal.Terminal;
-import org.jline.utils.InfoCmp;
+import org.jline.terminal.Terminal.Signal;
+import org.jline.terminal.Terminal.SignalHandler;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-// CHECKSTYLE_RULES.OFF: ClassDataAbstractionCoupling
 public abstract class Console implements Closeable {
-  // CHECKSTYLE_RULES.ON: ClassDataAbstractionCoupling
 
   private static final Logger log = LoggerFactory.getLogger(Console.class);
-  private static final List<String> PROPERTIES_COLUMN_HEADERS = Arrays.asList("Property", "Value");
+
+  private static final HandlerMap<KsqlEntity, Console> PRINT_HANDLERS =
+      HandlerMap.<KsqlEntity, Console>builder()
+          .put(CommandStatusEntity.class,
+              tablePrinter(CommandStatusEntity.class, CommandStatusTableBuilder::new))
+          .put(PropertiesList.class,
+              tablePrinter(PropertiesList.class, PropertiesListTableBuilder::new))
+          .put(Queries.class,
+              tablePrinter(Queries.class, QueriesTableBuilder::new))
+          .put(SourceDescriptionEntity.class,
+              (console, entity) -> console.printSourceDescription(entity.getSourceDescription()))
+          .put(SourceDescriptionList.class,
+              Console::printSourceDescriptionList)
+          .put(QueryDescriptionEntity.class,
+              (console, entity) -> console.printQueryDescription(entity.getQueryDescription()))
+          .put(QueryDescriptionList.class,
+              Console::printQueryDescriptionList)
+          .put(TopicDescription.class,
+              tablePrinter(TopicDescription.class, TopicDescriptionTableBuilder::new))
+          .put(StreamsList.class,
+              tablePrinter(StreamsList.class, StreamsListTableBuilder::new))
+          .put(TablesList.class,
+              tablePrinter(TablesList.class, TablesListTableBuilder::new))
+          .put(KsqlTopicsList.class,
+              tablePrinter(KsqlTopicsList.class, KsqlTopicsListTableBuilder::new))
+          .put(KafkaTopicsList.class,
+              tablePrinter(KafkaTopicsList.class, KafkaTopicsListTableBuilder::new))
+          .put(ExecutionPlan.class,
+              tablePrinter(ExecutionPlan.class, ExecutionPlanTableBuilder::new))
+          .put(FunctionNameList.class,
+              tablePrinter(FunctionNameList.class, FunctionNameListTableBuilder::new))
+          .put(FunctionDescriptionList.class,
+              Console::printFunctionDescription)
+          .build();
+
+  private static <T extends KsqlEntity> Handler<KsqlEntity, Console> tablePrinter(
+      final Class<T> entityType,
+      final Supplier<? extends TableBuilder<T>> tableBuilderType) {
+
+    try {
+      final TableBuilder<T> tableBuilder = tableBuilderType.get();
+
+      return (console, type) -> {
```

Comment:
nit: should `type` be called `ksqlEntity` instead?

Reply:
That's not a nit, it's a uber nit!